datadayessentials/config/_config.py

```diff
@@ -1,233 +1,161 @@
-                "database_credentials",
-                credentials_reference,
-            ]
-        )
+    def create_azure_app_configuration_client_connection(self, connection_string: str):
+        """
+        Creates a connection to Azure App Configuration.
 
-    @staticmethod
-    @catch_key_error
-    def get_database(database_reference: str) -> dict:
-        return LocalConfig.get_value_from_config(["databases", database_reference])
+        Args:
+            connection_string (str): The connection string for the Azure App Configuration.
 
-    @staticmethod
-    @catch_key_error
-    def list_available_databases() -> list:
-        return list(LocalConfig.get_value_from_config(["databases"]).keys())
+        Raises:
+            ConnectionError: If an error occurs while connecting to Azure.
+        """
+        try:
+            credential = DefaultAzureCredential()
+            credential = DataLakeAuthentication()
+            self.azure_app_config_connection_client = (
+                AzureAppConfigurationClient.from_connection_string(connection_string)
+            )
+        except Exception as e:
+            raise ConnectionError("An error occurred while connecting to Azure.") from e
 
-    @staticmethod
-    @catch_key_error
-    def get_local_cache_dir() -> str:
-        return os.path.join(Path.home(), LocalConfig().global_config["local_cache_dir"])
+    def client_azure_connection(self):
+        """
+        Manually connects to Azure.
 
-    @staticmethod
-    @catch_key_error
-    def get_data_lake_folder(
-        named_folder: str, data_lake: str = None, use_current_environment: bool = True
-    ) -> str:
-        """Retrieve a folder inside the data lake for the current environment, if use_current_environment is False then search across all envorinments for this named folder. The reason for this additional argument is because there are two use cases:
+        This method demonstrates a manual connection to Azure services.
+        It is a placeholder and should be replaced with actual implementation.
+        """
+        try:
+            # TODO: Implement function of manual authentication to azure
+            do_something()
+        except Exception as e:
+            raise ConnectionError("An error occurred while connecting to Azure via manual authentication.") from e
+
+    def set_environment_variables_from_cloud_config(self):
+        """
+        Retrieves values from Azure App Configuration and sets them as environment variables.
+        """
+        items = self.azure_app_config_connection_client.list_configuration_settings()
+        [self.set_environment_variable_as_per_environment(item.key, item.label, item.value) for item in items]
 
-        1. A folder that is environment specific (dev and prod have different locations for data storage)
-        2. A folder that is consistent across all environments
+    def set_environment_variable_as_per_environment(self, key, label, value):
```

Comment:
We need a dataclass here to store the variable names that are going to be dumped into the local config. For readability, this ensures that the reader knows what environment variables should be available in this environment. It also ensures we dont accidentally include environment variables that we dont need